flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -545,6 +551,19 @@ protected YarnClusterClient deployInternal() throws Exception {
 		// Set-up ApplicationSubmissionContext for the application
 		ApplicationSubmissionContext appContext = yarnApplication.getApplicationSubmissionContext();
 
+		final ApplicationId appId = appContext.getApplicationId();
```

Comment:
this line was only moved for my change. it is in fact used further down in the same method and cannot be moved inside the if.